src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -278,6 +280,13 @@ private Task ProcessStreamItem(HubConnectionContext connection, StreamItemMessag
                                     Log.StartingParameterStream(_logger, hubMethodInvocationMessage.StreamIds[streamPointer]);
                                     var itemType = descriptor.StreamingParameters[streamPointer];
                                     arguments[parameterPointer] = connection.StreamTracker.AddStream(hubMethodInvocationMessage.StreamIds[streamPointer], itemType);
+
+                                    // Check if we have an IAsyncEnumerable here and wrap the channel provided by AddStream
+                                    if (ReflectionHelper.IsIAsyncEnumerable(descriptor.OriginalParameterTypes[parameterPointer]))
+                                    {
```

Comment:
I’m not a fan of this. Everything on the sever should be in terms of IAsyncEnumerable not channel